src/transfer-manager.ts

```diff
@@ -0,0 +1,342 @@
+          passThroughOptionsCopy.destination || '',
+          file.name
+        );
+      }
+      if (options.stripPrefix) {
+        passThroughOptionsCopy.destination = file.name.replace(regex, '');
+      }
+      promises.push(limit(() => file.download(passThroughOptionsCopy)));
+    }
+
+    return Promise.all(promises);
+  }
+
+  /**
+   * @typedef {object} DownloadFileInChunksOptions
+   * @property {number} [concurrencyLimit] The number of concurrently executing promises
+   * to use when downloading the file.
+   * @property {number} [chunkSizeBytes] The size in bytes of each chunk to be downloaded.
+   * @experimental
+   */
+  /**
+   * Download a large file in chunks utilizing parallel download operations. This is a convenience method
+   * that utilizes {@link File#download} to perform the download.
+   *
+   * @param {object} [file | string] {@link File} to download.
+   * @param {DownloadFileInChunksOptions} [options] Configuration options.
+   * @returns {Promise<DownloadResponse>}
+   *
+   * @example
+   * ```
+   * const {Storage} = require('@google-cloud/storage');
+   * const storage = new Storage();
+   * const bucket = storage.bucket('my-bucket');
+   * const transferManager = new TransferManager(bucket);
+   *
+   * //-
+   * // Download a large file in chunks utilizing parallel operations.
+   * //-
+   * const response = await transferManager.downloadLargeFile(bucket.file('large-file.txt');
+   * // Your local directory now contains:
+   * // - "large-file.txt" (with the contents from my-bucket.large-file.txt)
+   * ```
+   * @experimental
+   */
+  async downloadFileInChunks(
+    fileOrName: File | string,
+    options: DownloadFileInChunksOptions = {}
+  ): Promise<void | DownloadResponse> {
+    let chunkSize =
+      options.chunkSizeBytes || DOWNLOAD_IN_CHUNKS_DEFAULT_CHUNK_SIZE;
+    let limit = pLimit(
+      options.concurrencyLimit || DEFAULT_PARALLEL_CHUNKED_DOWNLOAD_LIMIT
+    );
+    const promises = [];
+    const file: File =
+      typeof fileOrName === 'string'
+        ? this.bucket.file(fileOrName)
+        : fileOrName;
+
+    const fileInfo = await file.get();
+    const size = parseInt(fileInfo[0].metadata.size);
+    // If the file size does not meet the threshold download it as a single chunk.
+    if (size < DOWNLOAD_IN_CHUNKS_FILE_SIZE_THRESHOLD) {
+      limit = pLimit(1);
+      chunkSize = size;
+    }
+
+    let start = 0;
+    const filePath = options.destination || path.basename(file.name);
+    const fileToWrite = await fsp.open(filePath, 'w+');
+    while (start < size) {
+      const chunkStart = start;
+      let chunkEnd = start + chunkSize - 1;
+      chunkEnd = chunkEnd > size ? size : chunkEnd;
+      promises.push(
+        limit(() =>
+          file.download({start: chunkStart, end: chunkEnd}).then(resp => {
+            return fileToWrite.write(resp[0], 0, resp[0].length, chunkStart);
+          })
+        )
+      );
+
+      start += chunkSize;
+    }
```

Comment:
What's stopping the chunks from being written out of order?

Reply:
If I did this correctly the position in the file of where to write this chunk is handled by the `chunkStart` parameter. Basically this call should write the incoming chunk with no offset in the chunk buffer to the file offset specified by `chunkStart`. Was your understanding different?

Reply:
I've a slightly different understanding here (I could be wrong), but would it be possible for subsequent chunks to download before proceeding chunks and write to the file first?

Say we have a 65MB file and a 64MB chunk size (making 2 chunks to download). Since the 2nd chunk is 1MB and would download first, wouldn't it write to the file before the first chunk?

Reply:
It shouldn't matter if they are written in order. This implementation is leaning on a filesystem supporting [sparse files](https://en.wikipedia.org/wiki/Sparse_file). So, imagine you wrote these back-to-front; the first chunk will seek to its `chunkStart` near the end of the file, and the OS will logically pad that first byte with `0`s up to the seek point.

Reply:
^that's the missing piece for me, thanks Dom!

Reply:
As Dom outlined this was the behavior I was intending. It shouldn't matter which chunk downloads first they should get written to their correct place in the resultant file. Ultimately if all succeed, the file should be correctly "reassembled".